### PR TITLE
add support for creating a dcos-cli binary on linux

### PR DESCRIPTION
**Pull request by tamarrow**

### CODE REVIEW

File: binary/Dockerfile.linux-binary [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  jq \
+  make \
+  openssh-client \
+  git \
+  sudo \
+&& sudo apt-get update --fix-missing \
+&& sudo apt-get install -y python-dev build-essential \
+&& sudo apt-get install -y python-pip python-virtualenv \
+&& pip install pip --upgrade \
+&& pip install pyinstaller oauth2client
```
Comment (jsancio) [inline binary/Dockerfile.linux-binary]: We can make this part of the `requirement.txt` and we get it for free when do you do `make env`, no?

Comment (jsancio) [inline binary/Dockerfile.linux-binary]: Note, that I think we have two `requirement.txt` files. We probably want to make this part of the one in `dcos-cli/cli`.

File: Makefile [hunk 2]
```diff
@@ -14,3 +14,7 @@ doc: env
 
 packages: env
 	bin/packages.sh
+
+binary: clean env packages
+	rm -rf dist build
+	pyinstaller binary/binary.spec
```
Comment (jsancio) [inline Makefile]: I think we should put this in `cli/Makefile`. Technically we are turn the `dcoscli` package into a native binary.

File: binary/Dockerfile.linux-binary [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# This image runs the dcos-cli test suite.
```
Comment (jsancio) [inline binary/Dockerfile.linux-binary]: This should go in `cli/binary/Dockerfile.linux-binary` see previous comment.

File: binary/binary.spec [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# -*- mode: python -*-
```
Comment (jsancio) [inline binary/binary.spec]: Same here. We should move this to the `cli` folder (`dcoscli`).

File: cli/bin/test-binary.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash -e
+
+BASEDIR=`dirname $0`/..
+
+cd $BASEDIR
+
+PATH=$BASEDIR/../dist:$PATH
+source env/bin/activate
+cp tests/data/dcos.toml $DCOS_CONFIG
+py.test tests/integrations
```
Comment (jsancio) [inline cli/bin/test-binary.sh]: I think you can avoid the activate and deactivate by doing `env/bin/py.test tests/integrations`.

File: cli/Makefile [hunk 6]
```diff
@@ -9,6 +9,13 @@ env:
 test: env
 	bin/test.sh
 
+binary: clean env packages
+	rm -rf dist build
```
Comment (jsancio) [inline cli/Makefile]: Let's add this to the clean step/target. We can add it to the `bin/clean.sh` script.

File: cli/Makefile [hunk 7]
```diff
@@ -9,6 +9,12 @@ env:
 test: env
 	bin/test.sh
 
+binary: clean env packages
+	pyinstaller binary/binary.spec
+
+test-binary: env
```
Comment (jsancio) [inline cli/Makefile]: Shouldn't we depend on the `binary` target instead of `env`?

Comment (tamarrow) [inline cli/Makefile]: yep, updated.

File: cli/requirements.txt [hunk 8]
```diff
@@ -1,4 +1,6 @@
 sphinx>=1.3.1, <2.0
 tox>=2.2, <3.0
 wheel>=0.24.0, <1.0
+pytest>=2.9.1
+oauth2client>=1.5.2
```
Comment (jsancio) [inline cli/requirements.txt]: Why do we need this here?

**Comment by jsancio** [comment]
LGTM. One minor comment.

File: cli/requirements.txt (hunk 8 above)
Comment (tamarrow) [inline cli/requirements.txt]: Was needed when we had the original oauth, but we don't need it now! Removed it from install.py also.

**Pull request merged**